src/main/java/com/avispl/symphony/dal/avdevices/avcontrollers/dalite/screen/controller/DaliteScreenControllerCommunicator.java

```diff
@@ -0,0 +1,520 @@
+						presetNames.add(name);
+						addAdvancedControlProperties(advancedControllableProperty, controlStats,
+								createButton(group + DaLiteConstant.HASH + name, DaLiteConstant.RECALL, DaLiteConstant.RECALLING, 0L), DaLiteConstant.EMPTY);
+					}
+					break;
+				default:
+					logger.debug(String.format("the command %s doesn't support", command.getName()));
+					break;
+			}
+		}
+	}
+
+	/**
+	 * Populates the control buttons into the provided lists for control statistics and advanced controllable properties.
+	 * Adds buttons for system reboot, screen movement (up, down, stop) to the control map and advanced control properties.
+	 *
+	 * @param controlStats containing the current control statistics.
+	 * @param advancedControllableProperty to which the controls will be added.
+	 */
+	private void populateButtonControl(Map<String, String> controlStats, List<AdvancedControllableProperty> advancedControllableProperty) {
+		addAdvancedControlProperties(advancedControllableProperty, controlStats,
+				createButton(DaLiteConstant.SYSTEM + DaLiteConstant.HASH + DaLiteConstant.SYSTEM_REBOOT, DaLiteConstant.REBOOT, DaLiteConstant.REBOOTING, 240000L), DaLiteConstant.NA);
+		addAdvancedControlProperties(advancedControllableProperty, controlStats,
+				createButton(DaLiteConstant.SCREEN_CONTROL + DaLiteConstant.HASH + "MoveUp", "Up", "Moving", 0L), DaLiteConstant.NA);
+		addAdvancedControlProperties(advancedControllableProperty, controlStats,
+				createButton(DaLiteConstant.SCREEN_CONTROL + DaLiteConstant.HASH + "MoveDown", "Down", "Moving", 0L), DaLiteConstant.NA);
+		addAdvancedControlProperties(advancedControllableProperty, controlStats,
+				createButton(DaLiteConstant.SCREEN_CONTROL + DaLiteConstant.HASH + "MoveStop", "Stop", "Moving", 0L), DaLiteConstant.NA);
+	}
+
+	/**
+	 * Updates the current screen position value by sending the appropriate command and handling the response.
+	 * If successful, the screen position value is retrieved.
+	 */
+	private void updateScreenPositionValue() {
+		try {
+			sendCommandDetails(DaLiteCommand.SCREEN_POSITION.getCommand(), DaLiteCommand.SCREEN_POSITION.getName());
+		} catch (Exception e) {
+			logger.error("Error when get screen position", e);
+		}
+	}
+
+	/**
+	 * Populates a stats map with key-value pairs extracted from a response string using an enum.
+	 * Values are extracted from the response using the enum constant values.
+	 *
+	 * @param response the response string to parse.
+	 * @param stats the map to populate with extracted statistics.
+	 * @param enumClass the enum class containing constants with `getValue()` and `getName()` methods.
+	 * @param prefix optional prefix for keys in the stats map.
+	 * @param <E> the type of the enum.
+	 */
+	private <E extends Enum<E>> void populateStats(String response, Map<String, String> stats, Class<E> enumClass, String prefix) {
+		try {
+			for (E item : enumClass.getEnumConstants()) {
+				Method methodValue = item.getClass().getMethod("getValue");
```

Comment:
Is this necessary to use reflection for this?

Reply:
Yes, it's necessary for flexibility and maintainability.